### PR TITLE
fix: convert type of an option's value to string

### LINKED ISSUE #5157: [BUG] Mantine useSelect hook return value as type number instead of type string
### Describe the bug

For mantine Select, it only accepts value as string but the current useSelect hook return value type of number instead. It's kinda inconvenient to convert data type once again.

Current code of core useSelect doesn't actually convert the type of value:
Screen Shot 2023-10-24 at 12 47 38

Fix proposal:
a.toString() or String(a)

### Steps To Reproduce

 I reproduced the code: 
Screen Shot 2023-10-24 at 12 45 28
Screen Shot 2023-10-24 at 12 46 53

### Expected behavior

Each option should have `value` of type string

### Screenshot

_No response_

### Desktop

_No response_

### Mobile

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by an-tran511**
Currently, option value is not converted into string after getting result from `useSelect` hook.
I refactor to use String() constructor for type conversion instead of type casting

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

_Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request._

Please provide enough information so that others can review your pull request:

Explain the **details** for making this change. What existing problem does the pull request solve?

### Test plan (required)

Demonstrate the code is solid. If not, please add `WIP:` in its title.

### Closing issues

Put `closes #5157` in your comment to auto-close the issue that your PR fixes (if such).

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: .changeset/quick-knives-jump.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@refinedev/core": major
```
Comment (omeraplak) [inline .changeset/quick-knives-jump.md:2]:
```suggestion
"@refinedev/core": patch
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: .changeset/quick-knives-jump.md (hunk 1 above)
Comment (omeraplak) [reply .changeset/quick-knives-jump.md:2]: Thank you so much for your attention and effort! Can you mark the change as `patch` instead of `major`?

**Review by omeraplak — APPROVED** [review]

**Review by BatuhanW — COMMENTED** [review]
Hello @an-tran511 

Thanks for the contribution! Looks good to me.

Before we merge it, could you add some test cases to cover your fix [in this file?](https://github.com/refinedev/refine/blob/90da1f82bba95edcf09a46a7fb51b760e4c711f2/packages/core/src/hooks/useSelect/index.spec.ts)

**Review by alicanerdurmaz — APPROVED** [review]

**Comment by an-tran511** [comment]
> Hello @an-tran511
> 
> Thanks for the contribution! Looks good to me.
> 
> Before we merge it, could you add some test cases to cover your fix [in this file?](https://github.com/refinedev/refine/blob/90da1f82bba95edcf09a46a7fb51b760e4c711f2/packages/core/src/hooks/useSelect/index.spec.ts)

I added test case for this patch

**Review by BatuhanW — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

File: .changeset/quick-knives-jump.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@refinedev/core": patch
+---
+
+convert type of an option's value to string
```
Comment (aliemir) [inline .changeset/quick-knives-jump.md:5]: fix: convert type of an option's `value` to `string`

**Pull request merged**